### DISCUSSION TITLE
'handling head updates in window post sched: handler: websocket connection closed "What is the problem? It occurs every few days, and then the encapsulation is stuck in" waitspeed ", but not" SubmitWindowedPoSt "?

### DISCUSSION THREAD

**Discussion by ChaiWenyu1983** [filecoin-project/lotus#9551] [Storage Provider]
`{"level":"error","ts":"2022-10-25T05:38:13.755+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:221","msg":"handling head updates in window post sched: handler: websocket connection closed"}
{"level":"warn","ts":"2022-10-25T05:38:13.757+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:164","msg":"window post scheduler notifs channel closed"}
{"level":"info","ts":"2022-10-25T11:58:15.901+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:158","msg":"restarting window post scheduler"}
{"level":"warn","ts":"2022-10-25T12:11:00.408+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":3,"good":0}
{"level":"warn","ts":"2022-10-25T12:11:00.408+0800","logger":"wdpost","caller":"wdpost/wdpost_run_faults.go:147","msg":"No recoveries to declare","deadline":0,"faulty":3}
{"level":"warn","ts":"2022-10-25T13:10:59.770+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":1,"good":0}
{"level":"warn","ts":"2022-10-25T13:10:59.771+0800","logger":"wdpost","caller":"wdpost/wdpost_run_faults.go:147","msg":"No recoveries to declare","deadline":2,"faulty":1}
{"level":"warn","ts":"2022-10-25T13:14:04.917+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":1022,"good":1022}
{"level":"info","ts":"2022-10-25T13:14:04.956+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:388","msg":"running window post","chain-random":"qETl8Tlo9/vz2RPdiMjnD9eAnUimuIFgfIkrLqk+pKI=","deadline":{"CurrentEpoch":2278942,"PeriodStart":2278961,"Index":0,"Open":2278961,"Close":2279021,"Challenge":2278941,"FaultCutoff":2278891,"WPoStPeriodDeadlines":48,"WPoStProvingPeriod":2880,"WPoStChallengeWindow":60,"WPoStChallengeLookback":20,"FaultDeclarationCutoff":70},"height":"2278942","skipped":0}
{"level":"warn","ts":"2022-10-25T13:32:03.373+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:164","msg":"window post scheduler notifs channel closed"}
{"level":"error","ts":"2022-10-25T13:32:13.511+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:151","msg":"ChainNotify error: handler: websocket connection closed"}
{"level":"info","ts":"2022-10-25T13:32:32.775+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:158","msg":"restarting window post scheduler"}
{"level":"warn","ts":"2022-10-25T13:34:00.741+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:164","msg":"window post scheduler notifs channel closed"}
{"level":"info","ts":"2022-10-25T13:38:24.004+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:158","msg":"restarting window post scheduler"}
{"level":"warn","ts":"2022-10-25T13:38:24.014+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":1,"good":0}
{"level":"warn","ts":"2022-10-25T13:38:24.014+0800","logger":"wdpost","caller":"wdpost/wdpost_run_faults.go:147","msg":"No recoveries to declare","deadline":2,"faulty":1}
{"level":"warn","ts":"2022-10-25T13:38:24.155+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:164","msg":"window post scheduler notifs channel closed"}
{"level":"info","ts":"2022-10-25T13:39:42.711+0800","logger":"wdpost","caller":"wdpost/wdpost_sched.go:158","msg":"restarting window post scheduler"}
{"level":"warn","ts":"2022-10-25T13:39:42.716+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":1,"good":0}
{"level":"warn","ts":"2022-10-25T13:39:42.716+0800","logger":"wdpost","caller":"wdpost/wdpost_run_faults.go:147","msg":"No recoveries to declare","deadline":2,"faulty":1}
{"level":"warn","ts":"2022-10-25T13:43:07.883+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":1022,"good":1022}
{"level":"info","ts":"2022-10-25T13:43:07.925+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:388","msg":"running window post","chain-random":"qETl8Tlo9/vz2RPdiMjnD9eAnUimuIFgfIkrLqk+pKI=","deadline":{"CurrentEpoch":2278999,"PeriodStart":2278961,"Index":0,"Open":2278961,"Close":2279021,"Challenge":2278941,"FaultCutoff":2278891,"WPoStPeriodDeadlines":48,"WPoStProvingPeriod":2880,"WPoStChallengeWindow":60,"WPoStChallengeLookback":20,"FaultDeclarationCutoff":70},"height":"2278999","skipped":0}
{"level":"info","ts":"2022-10-25T13:54:40.008+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:403","msg":"computing window post","batch":0,"elapsed":692.083287197,"skip":0,"err":null}
{"level":"warn","ts":"2022-10-25T14:11:32.832+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:234","msg":"Checked sectors","checked":34,"good":34}
{"level":"info","ts":"2022-10-25T14:11:32.836+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:388","msg":"running window post","chain-random":"MxToXE7gfCSjY+r0LvTMeJU0NM1rBgZglY9sxMAYrh4=","deadline":{"CurrentEpoch":2279063,"PeriodStart":2278961,"Index":2,"Open":2279081,"Close":2279141,"Challenge":2279061,"FaultCutoff":2279011,"WPoStPeriodDeadlines":48,"WPoStProvingPeriod":2880,"WPoStChallengeWindow":60,"WPoStChallengeLookback":20,"FaultDeclarationCutoff":70},"height":"2279063","skipped":0}
{"level":"info","ts":"2022-10-25T14:21:05.695+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:403","msg":"computing window post","batch":0,"elapsed":572.858769787,"skip":0,"err":null}
{"level":"info","ts":"2022-10-25T14:22:30.704+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:619","msg":"Submitted window post: bafy2bzacec3rkbhmghwqbx6wzeu2yhylsqgdmkwqgnvpfekevkkyxpe4uoeee (deadline 2)"}
{"level":"info","ts":"2022-10-25T14:26:00.266+0800","logger":"wdpost","caller":"wdpost/wdpost_run.go:629","msg":"Window post submission successful","cid":"bafy2bzacec3rkbhmghwqbx6wzeu2yhylsqgdmkwqgnvpfekevkkyxpe4uoeee","deadline":2,"epoch":"2279087","ts":[{"/":"bafy2bzaceaeokcbalqrw67vo7tswtccl32jqwde6sv7xwa7brdhrgvtx7qqda"},{"/":"bafy2bzacea24p64t7hilugdtnzrxb6qcepslix7mkrwjofg7c6aqwtjafozoq"},{"/":"bafy2bzaced7mdcc2non5u26reqxer2opgyuorjhs6um47l3zropeand4xhq4k"},{"/":"bafy2bzacea7euwtpnelxdmmbmqwxpowrawknshzizlqjcqvgjjrmlzhbv5n4k"},{"/":"bafy2bzaceduc4bktlisermefjhaa5vvg25cxovh6q47n4iwysnwrqhmb2w44m"},{"/":"bafy2bzacea2676kp7wb33eincw3cdyj4n7wpdizkysrvr4xzg4liifgtb33e2"},{"/":"bafy2bzacebylwaxynrkjm5futb3xcd6ei5biscbloeic5aykd2zl5anxch7kq"},{"/":"bafy2bzacecrwwajr2p764ynf6xbnzhb2t5jwbypk2mo3awouhybjqkus6dau2"}]}
`

[miner.log](https://github.com/filecoin-project/lotus/files/9859546/miner.log)

**Comment by Reiers** [comment]
https://github.com/filecoin-project/lotus/issues/8362
Feel free to join the conversation on the issue ticket.

>  It occurs every few days, and then the encapsulation is stuck in" waitspeed ", but not" SubmitWindowedPoSt "?
- Could you rephrase the question? is the sectors stuck in `WaitSeed` and you have wdPost failure?

**Reply by ChaiWenyu1983** [reply]
The "winpost" is also abnormal. At the same time, the encapsulation is stuck in the "waitspeed". After the "miner" service is restarted, it is normal. It occurs regularly, almost once a week.
